### PR TITLE
Sigint trap

### PR DESCRIPTION
**Pull request by mdsauce**

### CODE REVIEW

**Comment by mdsauce** [comment]
CTRL-C gracefully shuts down tunnels.  No hardkill or Double Ctrl-c at the moment

**Comment by mdsauce** [comment]
closes #26

File: cmd/start.go [hunk 1]
```diff
@@ -63,17 +65,22 @@ var startCmd = &cobra.Command{
...
+				pool := <-metadata
 				logger.Disklog.Debugf("%s pool is %s.  Metadata is %v", fscanner.Text(), pool, meta[pool])
 				go manager.Start(fscanner.Text(), &wg, meta[pool])
 			}
 		}
-		// TODO catch the ctrl-c here if it comes.
+		signal.Notify(stop, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+		go func() {
+			<-stop
+			logger.Disklog.Warn("User pressed ctrl-c. Hard killing tunnels now.  Active jobs using these tunnels will die.")
```
Comment (enriquegh) [inline cmd/start.go:82]: Is this true? I started 3 HA tunnels and started a live test. While test was still running, I sent a SIGINT to sauced.
Two tunnels shutdown and the third one waited until the process was finished.

File: manager/tunnel_unix.go [hunk 2]
```diff
@@ -114,5 +115,6 @@ func StopAll() {
 	last := GetLastKnownState()
 	for _, tunnel := range last.Tunnels {
 		Stop(tunnel.PID)
+		// add a stop via REST API func here.  So rest api gets signal as well.
```
Comment (enriquegh) [inline manager/tunnel_unix.go:118]: Is this needed? If we're sending a SIGINT to the SC client process, the client itself should be sending that already.

File: cmd/start.go (hunk 1 above)
Comment (mdsauce) [reply cmd/start.go:82]: now works!  Try shutting down running tunnels with a ctrl-c to sauced.  they will gracefully shutdown

**Pull request merged**